jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/Histogram.java

```diff
@@ -0,0 +1,245 @@
+                    for (int j = 0; j < 256; ++j) {
+                        histogram[i][j] /= yScale;
+                    }
+                }
+            }
+        }
+
+        maxV = histogram[3][0];
+        for (int i = 1; i < 256; ++i) {
+            if (histogram[3][i] > maxV) {
+                maxV = histogram[3][i];
+            }
+        }
+        double alphaScale = maxV;
+        alphaScale /= (outputHeight - 2 * TOPMARGIN);
+        if( alphaScale != 0 ){
+            for (int i = 0; i < 256; ++i) {
+                histogram[3][i] /= alphaScale;
+            }
+        }
+    }
+
+
+    private void drawHistogram(){
+
+        int barWidth = outputWidth / 256;
+        int leftmargin = outputWidth - 256*barWidth;
+        leftmargin /= 2;
+        int rightmargin = leftmargin;
+        if(( leftmargin * 2 ) != ( outputWidth - 256*barWidth )){
+            leftmargin++;
+        }
+
+        int index = 0;
+        for(int i = leftmargin ; i < outputWidth - rightmargin ; ++i){
+            for(int j = TOPMARGIN ; j < outputHeight - TOPMARGIN ; ++j){
+                int color = 0xFF000000;
+                if( blackAndWhite ){
+                    if( blackAndWhiteHistogram[index/barWidth] >= outputHeight - TOPMARGIN - j ){
+                        color += 0xb0b0b0;
+                    }
+                } else {
+                    if( histogram[0][index/barWidth] >= outputHeight - TOPMARGIN - j ){
+                        color += 0xc80000;
+                    }
+                    if( histogram[1][index/barWidth] >= outputHeight - TOPMARGIN - j ){
+                        color += 0x00c800;
+                    }
+                    if( histogram[2][index/barWidth] >= outputHeight - TOPMARGIN - j ){
+                        color += 0x0000c8;
+                    }
+                    if( color == 0xFFc8c8c8 ){
+                        color = 0xFFb0b0b0;
+                    }
+                    if( histogram[3][index/barWidth] >= outputHeight - TOPMARGIN - j ){
+                        color += 0x323232;
+                    }
+                }
+                output.setRGB(i, j, color);
+            }
+            index++;
+        }
+    }
+
+    private void setChannels(ColorChannel[] channels){
+        for(ColorChannel c : channels){
+            if( c.equals(ColorChannel.RED) ){
+                channel[0] = true;
+            }
+            if( c.equals(ColorChannel.GREEN) ){
+                channel[1] = true;
+            }
+            if( c.equals(ColorChannel.BLUE) ){
+                channel[2] = true;
+            }
+            if( c.equals(ColorChannel.ALPHA) ){
+                channel[3] = true;
+            }
+        }
+    }
+
+    private boolean isAlphaIdentical(){
+        int moreThanZero = 0;
+        for(int i = 0 ; i < 256 ; ++i){
+            if( histogram[ 3 ][ i ] > 0 ){
+                moreThanZero++;
+            }
+        }
+        return moreThanZero <= 1;
```

Comment:
what is this even supposed to do? it will return true for both 1 and 0, variable is named moreThanZero?